scss/_patterns_rule.scss

```diff
@@ -0,0 +1,20 @@
+@import 'settings';
+
+$small-offset: map-get($grid-margin-widths, small);
+$default-offset: map-get($grid-margin-widths, default);
+
+@mixin vf-p-rule {
+  .p-rule {
+    @extend %hr;
+    @extend %u-no-margin--bottom--hr;
+    max-width: calc($grid-max-width - 2 * $small-offset);
```

Comment:
This is not working well unfortunately.

It's fine on large screens, bigger than max-width:

<img width="1159" alt="image" src="https://user-images.githubusercontent.com/83575/234544948-43cfb781-44a3-4a64-b272-3971a72f7e43.png">

But on a screens smaller than max-width, it still takes the whole width of the screen (because it is smaller than max width), while at the same time grid has padding:

<img width="1078" alt="image" src="https://user-images.githubusercontent.com/83575/234545182-24ba6032-72b8-4ce6-8281-a96d3a1b668f.png">


This makes it trickier to implement, because we need to take that into account into width, similar to what `is-fixed-width` variant of current `hr` does:

https://github.com/canonical/vanilla-framework/blob/main/scss/_base_hr.scss#L20-L40

But, it needs to happen ONLY if it's outside of the grid containers. If it is inside row. Which `is-fixed-width` also takes into account.

So I guess porting whole `is-fixed-width` functionality into `p-rule` should solve it hopefully?

Reply:
@bartaz so why not just put that in the hr directly? the code is already there.

also what ius $small-offset? this sounds very cryptic

Reply:
> @bartaz so why not just put that in the hr directly? the code is already there.

I guess you mean using the styles of `is-fixed-width` directly on `<hr>` element by default, without a class name? That could work. My only worry is if someone is using `<hr>` outside of standard grid environment (like application layout). it may not be expected to have max width and auto margins.

This also reminds me… In the new Vanilla layout (that we have on design site and Vanilla), it will also not work out of the box, because it's not a centered standard grid anymore. You would need to put it inside strip or row anyway…